### DISCUSSION TITLE
Pyinstaller doesn't pick up `selectolax` ?

### DISCUSSION THREAD

**Discussion by baseplate-admin** [pyinstaller/pyinstaller#8703] [Help] [answered]
Hi, after building a package containing [`selectolax`](https://github.com/rushter/selectolax), it seems that i have this error.

```powershell
PyInstaller\hooks\rthooks\pyi_rth_pkgres.py:36: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
Traceback (most recent call last):
  File "cli.py", line 2, in <module>
  File "builder\staff.py", line 2, in <module>
  File "selectolax\__init__.py", line 10, in <module>
ImportError: cannot import name 'modest' from partially initialized module 'selectolax' (most likely due to a circular import) (C:\Users\baseplate-admin\AppData\Local\Temp\_MEI191562\selectolax\__init__.pyc)
[2272] Failed to execute script 'cli' due to unhandled exception!
```

And i have this spec file
```python
# -*- mode: python ; coding: utf-8 -*-


a = Analysis(
    ['src\\shiinobi\\cli.py'],
    pathex=['src\\shiinobi'],
    binaries=[],
    datas=[],
    hiddenimports=['selectolax'],
    hookspath=[],
    hooksconfig={},
    runtime_hooks=[],
    excludes=[],
    noarchive=True,
    optimize=2,
)
pyz = PYZ(a.pure)

exe = EXE(
    pyz,
    a.scripts,
    a.binaries,
    a.datas,
    [],
    name='__main__',
    debug=False,
    bootloader_ignore_signals=False,
    strip=True,
    upx=True,
    upx_exclude=[],
    runtime_tmpdir=None,
    console=True,
    disable_windowed_traceback=False,
    argv_emulation=False,
    target_arch=None,
    codesign_identity=None,
    entitlements_file=None,
)
```

**Comment by rokm** [comment, marked as answer]
Looks like `selectolax.modest` is not collected, and at least in the latest version of `selectolax`, it seems to be a namespace package without any modules (it's a directory with `.pxi` files). 

Can you try adding `selectolax.modest` to `hiddenimports`? 

Alternatively, collecting data files from `selectolax` might also do the trick (add `from PyInstaller.utils.hooks import collect_data_files` and then do `datas=collect_data_files('selectolax')`), because it will ensure that `selectolax/modest` directory exists on filesystem.

**Reply by baseplate-admin** [reply]
>collecting data files from selectolax might also do the trick (add from PyInstaller.utils.hooks import collect_data_files and then do datas=collect_data_files('selectolax')), because it will ensure that selectolax/modest directory exists on filesystem.


Thanks. it works

**Reply by baseplate-admin** [reply]
Hey @rokm, i have a small question, can i contribute this hook back to `pyinstaller`?

**Reply by rokm** [reply]
Sure! It should go into https://github.com/pyinstaller/pyinstaller-hooks-contrib, though.
